### PR TITLE
execute docker pull before running container

### PR DESCRIPTION
**Pull request by Rob-Johnson**
We found an edge case where this failed because the image didn't already exist on the host travis was running the tests on - dockerpy doesn't pull an image if it doesn't already exist, so this addresses that.

### CODE REVIEW

File: general_itests/steps/paasta_execute_docker_command.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ def create_docker_container(context, task_id):
         context.docker_client.remove_container(container_name, force=True)
     except APIError:
         pass
+    context.docker_client.pull('ubuntu:trusty')
```
Comment (mrtyler) [inline general_itests/steps/paasta_execute_docker_command.py]: Uh, do you have to hardcode this name? This seems pretty bad.

**Comment by solarkennedy** [comment]
ship

File: general_itests/steps/paasta_execute_docker_command.py (hunk 1 above)
Comment (Rob-Johnson) [inline general_itests/steps/paasta_execute_docker_command.py]: hm, thats fair, I could change the feature so that the image name is parameterised.

**Comment by Rob-Johnson** [comment]
r2: parameterised image name

**Comment by mrtyler** [comment]
I didn't notice this was only in itest code but probably still a good idea :)

shipit
